src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -217,18 +218,32 @@ public function extractSessionContainer(Token $token = null) : Data
     /**
      * @param Data     $sessionContainer
      * @param Response $response
+     * @param Token    $originalToken
      *
      * @return Response
      *
      * @throws \InvalidArgumentException
      */
-    private function appendToken(Data $sessionContainer, Response $response) : Response
+    private function appendToken(Data $sessionContainer, Response $response, Token $originalToken = null) : Response
     {
-        if ($sessionContainer->isEmpty()) {
+        $empty = $sessionContainer->isEmpty();
+
+        if ($empty && null === $originalToken) {
```

Comment:
We cannot remove the session cookie from the response headers, we actually need to inform the client that it is now expired and shouldn't be sent again. Right?